pkg/report/table/table_private_test.go

```diff
@@ -0,0 +1,232 @@
+var (
+	osVuln = types.Result{
+		Target: "test (alpine 3.20.3)",
+		Class:  types.ClassOSPkg,
+		Type:   ftypes.Alpine,
+		Vulnerabilities: []types.DetectedVulnerability{
+			{
+				VulnerabilityID: "CVE-2024-9143",
+				PkgName:         "libcrypto3",
+			},
+			{
+				VulnerabilityID: "CVE-2024-9143",
+				PkgName:         "libssl3",
+			},
+		},
+	}
+	jarVuln = types.Result{
+		Target: "Java",
+		Class:  types.ClassLangPkg,
+		Type:   ftypes.Jar,
+		Vulnerabilities: []types.DetectedVulnerability{
+			{
+				VulnerabilityID: "CVE-2022-42003",
+				PkgName:         "com.fasterxml.jackson.core:jackson-databind",
+				PkgPath:         "app/jackson-databind-2.13.4.1.jar",
+			},
+			{
+				VulnerabilityID: "CVE-2021-44832",
+				PkgName:         "org.apache.logging.log4j:log4j-core",
+				PkgPath:         "app/log4j-core-2.17.0.jar",
+			},
+		},
+	}
+
+	noVuln = types.Result{
+		Target: "requirements.txt",
+		Class:  types.ClassLangPkg,
+		Type:   ftypes.Pip,
+	}
+
+	dockerfileMisconfig = types.Result{
+		Target: "app/Dockerfile",
+		Class:  types.ClassConfig,
+		Type:   ftypes.Dockerfile,
+		Misconfigurations: []types.DetectedMisconfiguration{
+			{
+				ID: "DS002",
+			},
+			{
+				ID: "DS026",
+			},
+		},
+	}
+	secret = types.Result{
+		Target: "requirements.txt",
+		Class:  types.ClassSecret,
+		Secrets: []types.DetectedSecret{
+			{
+				RuleID: "aws-access-key-id",
+			},
+		},
+	}
+	osLicense = types.Result{
+		Target: "OS Packages",
+		Class:  types.ClassLicense,
+		Licenses: []types.DetectedLicense{
+			{
+				Name: "GPL-2.0-only",
+			},
+		},
+	}
+
+	jarLicense = types.Result{
+		Target: "Java",
+		Class:  types.ClassLicense,
+	}
+	fileLicense = types.Result{
+		Target: "Loose File License(s)",
+		Class:  types.ClassLicenseFile,
+	}
+)
+
+func Test_renderSummary(t *testing.T) {
+	tests := []struct {
+		name           string
+		scanners       types.Scanners
+		noSummaryTable bool
+		report         types.Report
+		want           string
+	}{
+		{
+			name: "happy path all scanners",
+			scanners: []types.Scanner{
+				types.VulnerabilityScanner,
+				types.MisconfigScanner,
+				types.SecretScanner,
+				types.LicenseScanner,
+			},
+			report: types.Report{
+				Results: []types.Result{
+					osVuln,
+					jarVuln,
+					dockerfileMisconfig,
+					secret,
+					osLicense,
+					jarLicense,
+					fileLicense,
+				},
+			},
+			want: `
+Report Summary
+
+┌───────────────────────┬────────────┬─────────────────┬───────────────────┬─────────┬──────────┐
+│        Target         │    Type    │ Vulnerabilities │ Misconfigurations │ Secrets │ Licenses │
+├───────────────────────┼────────────┼─────────────────┼───────────────────┼─────────┼──────────┤
+│ test (alpine 3.20.3)  │   alpine   │        2        │         -         │    -    │    -     │
+├───────────────────────┼────────────┼─────────────────┼───────────────────┼─────────┼──────────┤
+│ Java                  │    jar     │        2        │         -         │    -    │    -     │
```

Comment:
The summary table contains aggregated targets.
Do we want to split them by `filePath` in this PR or just start working on removing the aggregation (in another PR)?
cc. @knqyf263

Reply:
I didn't split aggregated results, but, actually yes, we can split them in the table writer. I think we should do that in this PR. We will not need the logic after removing the aggregation, though.

Reply:
Done - https://github.com/aquasecurity/trivy/pull/8177/commits/84169b4fc9922ebb285938dc0403b16ef5680764 + https://github.com/aquasecurity/trivy/pull/8177/commits/892d6de100872ccb3e9f2019366a4555c86176c9